### PR TITLE
[SUBMARINE-208] Create Architecture and Requirement Doc

### PR DESCRIPTION
**Pull request by wangdatan**
### What is this PR for?
Design doc for architecture and requirement

### What type of PR is it?
Design Documentation.

### Todos
Several TODOs remain in the doc.

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-208

### CODE REVIEW

File: docs/design/Architecture and Requirement.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
+# Terminology
```
Comment (liuxunorg) [inline docs/design/Architecture and Requirement.md]: Need add Apache license head.

File: docs/design/Architecture and Requirement.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+It is a complex workflow from raw data to usable models, after talking to many different data scientists, we have learned that a typical procedure to train a new model and push to production can take months to 1-2 years. 
+
+It is also a wide skill set required by this workflow. For example, data transformation needs tools like Spark/Hive for large scale and tools like Pandas for small scale. And model training needs to be switched between XGBoost, Tensorflow, Keras, PyTorch. Building a data pipeline needs Apache Airflow or Oozie. 
+
+Yes, there are great, standardized open-source tools built for many of such purposes. But how about changes need to be made for a particular part of the data pipeline? How about adding a few columns to the training data for experiments? How about training models, and push models to validation, A/B testing before rolling to production? All these steps need jumping between different tools, UIs, and very hard to make changes, and it is not error-proof during these procedures.
+
+#### Pain \#2 Dependencies of underlying resource management platform
```
Comment (liuxunorg) [inline docs/design/Architecture and Requirement.md]: Document heading level error.
Need change `#### Pain` to `### Pain`

**Comment by liuxunorg** [comment]
hi @wangdatan 
1. You need to add `[SUBMARINE-208]` or `SUBMARINE-208. ` in front of the  PR title. `merge-submarine-pr.sh` requires this association JIRA.
2. You need to modify this paragraph in the PR
```
What is the Jira issue?
SUBMARINE-208
```
to
```
What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-208
```
In this way, the JIRA jump URL will be available in the merged commit log.

**Comment by tangzhankun** [comment]
Thanks @wangdatan for making this design. I'm thinking whether we need to consider inference at present. Or maybe we can add this later in the design doc. It looks good to me. +1.

File: docs/design/Architecture and Requirement.md (hunk 1 above)
Comment (wangdatan) [reply docs/design/Architecture and Requirement.md]: updated

File: docs/design/Architecture and Requirement.md (hunk 2 above)
Comment (wangdatan) [reply docs/design/Architecture and Requirement.md]: updated

**Comment by wangdatan** [comment]
Addressed all comments. Please check it again, @liuxunorg , @tangzhankun

**Comment by liuxunorg** [comment]
LGTM

File: docs/design/architecture-and-requirements.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+| -------- | -------- |
+| User | A single data-scientist/data-engineer. User has resource quota, credentials |
+| Team | User belongs to one or more teams, teams have ACLs for artifacts sharing such as notebook content, model, etc. |
+| Admin | Also called SRE, who manages user's quotas, credentials, team, and other components. |
+| Project | A project may include one or multiple notebooks, zero or multiple running jobs. And could be collaborated by multiple users who have ACLs on it |
+
+
+# Background 
+
+Everybody talks about machine learning today, and lots of companies are trying to leverage machine learning to push the business to the next level. Nowadays, as more and more developers, infrastructure software companies coming to this field, machine learning becomes more and more archivable.
```
Comment (jojochuang) [inline docs/design/architecture-and-requirements.md]: "archivable" --> achievable.

File: docs/design/architecture-and-requirements.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+
+# Submarine
+
+## Overview
+
+### A little bit history
+
+Initially, Submarine is built to solve problems of running deep learning jobs like Tensorflow/PyTorch on Apache Hadoop YARN, allows admin to monitor launched deep learning jobs, and manage generated models. 
+
+It was part of YARN initially, code resides under `hadoop-yarn-applications`. Later, the community decided to move to a subject of Hadoop because we want to support other resource management platforms like K8s. And finally, we're reconsidering Submarine's charter and Hadoop community voted that it is the time to moved Submarine to a separate Apache TLP.
```
Comment (jojochuang) [inline docs/design/architecture-and-requirements.md]: "move to a subject of Hadoop" --> not clear what exactly this means. "convert it to be a subproject within Hadoop"?

File: docs/design/architecture-and-requirements.md (hunk 3 above)
Comment (wangdatan) [reply docs/design/architecture-and-requirements.md]: Addressed.

File: docs/design/architecture-and-requirements.md (hunk 4 above)
Comment (wangdatan) [reply docs/design/architecture-and-requirements.md]: Addressed.

**Pull request closed without merging**